frontstage/views/register/activate_account.py

```diff
@@ -15,22 +14,24 @@
 
 @register_bp.route('/activate-account/<token>', methods=['GET'])
 def register_activate_account(token):
-    logger.info('Attempting to verify email')
-    response = api_call('PUT', app.config['VERIFY_EMAIL'], parameters={'token': token})
-
-    # Handle api errors
-    if response.status_code == 409:
-        logger.info('Expired email verification token', token=token)
-        return render_template('register/register.link-expired.html')
-    elif response.status_code == 404:
-        logger.warning('Unrecognised email verification token', token=token)
-        return redirect(url_for('error_bp.not_found_error_page'))
-    elif response.status_code != 200:
-        logger.info('Failed to verify email')
-        raise ApiError(response)
+    logger.info('Attempting to verify email', token=token)
+
+    try:
+        party_controller.verify_email(token)
+    except ApiError as exc:
+        # Handle api errors
+        if exc.status_code == 409:
+            logger.info('Expired email verification token', token=token)
+            return render_template('register/register.link-expired.html')
+        elif exc.status_code == 404:
+            logger.warning('Unrecognised email verification token', token=token)
+            return redirect(url_for('error_bp.not_found_error_page'))
+        elif exc.status_code != 200:
```

Comment:
Should an ApiError ever have a 200 status code?

Reply:
Will change to an `else`.